File: src/main/java/voidchess/player/ki/ConstantEvaluation.java
package voidchess.player.ki;

import voidchess.board.ChessGameInterface;
import voidchess.player.ki.evaluation.Evaluated;
import voidchess.player.ki.evaluation.Draw;

/**
 * usefull for Profile purposes (only DynamicEvaluation remains)
 */

public class ConstantEvaluation implements StaticEvaluationInterface {
    @Override
    public Evaluated getPrimaryEvaluation(ChessGameInterface game, boolean forWhite) {
        return Draw.INSTANCE;
    }

    @Override
    public void addSecondaryEvaluation(ChessGameInterface game, boolean forWhite, Evaluated withPrimaryEvaluation) {
        //do nothing (keep the secondary value = 0)
    }
}

File: src/main/kotlin/voidchess/player/ki/ConstantEvaluation.kt
package voidchess.player.ki

import voidchess.board.ChessGameInterface
import voidchess.player.ki.evaluation.Evaluated
import voidchess.player.ki.evaluation.Ongoing

/**
 * usefull for Profile purposes (only DynamicEvaluation remains)
 */

class ConstantEvaluation : StaticEvaluationInterface {

    override fun getPrimaryEvaluation(game: ChessGameInterface, forWhite: Boolean) = nullValue

    override fun addSecondaryEvaluation(game: ChessGameInterface, forWhite: Boolean, withPrimaryEvaluation: Evaluated) {
        //do nothing (keep the secondary value = 0)
    }

    companion object {
        private val nullValue = Ongoing(0.0)
    }
}
